common/flatpak-dir-utils.c

```diff
@@ -0,0 +1,586 @@
+      return NULL;
+    }
+
+  if (dir_out)
+    *dir_out = g_object_ref (dir);
+  return g_steal_pointer (&deploy);
+}
+
+GFile *
+flatpak_find_files_dir_for_ref (FlatpakDecomposed *ref,
+                                GCancellable      *cancellable,
+                                GError           **error)
+{
+  g_autoptr(GFile) deploy = NULL;
+
+  deploy = flatpak_find_deploy_dir_for_ref (ref, NULL, cancellable, error);
+  if (deploy == NULL)
+    return NULL;
+
+  return g_file_get_child (deploy, "files");
+}
+
+GFile *
+flatpak_find_unmaintained_extension_dir_if_exists (const char   *name,
+                                                   const char   *arch,
+                                                   const char   *branch,
+                                                   GCancellable *cancellable)
+{
+  g_autoptr(FlatpakDir) user_dir = NULL;
+  g_autoptr(GFile) extension_dir = NULL;
+  g_autoptr(GError) local_error = NULL;
+
+  user_dir = flatpak_dir_get_user ();
+
+  extension_dir = flatpak_dir_get_unmaintained_extension_dir_if_exists (user_dir, name, arch, branch, cancellable);
+  if (extension_dir == NULL)
+    {
+      g_autoptr(GPtrArray) system_dirs = NULL;
+      int i;
+
+      system_dirs = flatpak_dir_get_system_list (cancellable, &local_error);
+      if (system_dirs == NULL)
+        {
+          g_warning ("Could not get the system installations: %s", local_error->message);
+          return NULL;
+        }
+
+      for (i = 0; i < system_dirs->len; i++)
+        {
+          FlatpakDir *system_dir = g_ptr_array_index (system_dirs, i);
+          extension_dir = flatpak_dir_get_unmaintained_extension_dir_if_exists (system_dir, name, arch, branch, cancellable);
+          if (extension_dir != NULL)
+            break;
+        }
+    }
+
+  if (extension_dir == NULL)
+    return NULL;
+
+  return g_steal_pointer (&extension_dir);
+}
+
+FlatpakDecomposed *
+flatpak_find_current_ref (const char   *app_id,
+                          GCancellable *cancellable,
+                          GError      **error)
+{
+  g_autoptr(FlatpakDecomposed) current_ref = NULL;
+  g_autoptr(FlatpakDir) user_dir = flatpak_dir_get_user ();
+  int i;
+
+  current_ref = flatpak_dir_current_ref (user_dir, app_id, NULL);
+  if (current_ref == NULL)
+    {
+      g_autoptr(GPtrArray) system_dirs = NULL;
+
+      system_dirs = flatpak_dir_get_system_list (cancellable, error);
+      if (system_dirs == NULL)
+        return FALSE;
+
+      for (i = 0; i < system_dirs->len; i++)
+        {
+          FlatpakDir *dir = g_ptr_array_index (system_dirs, i);
+          current_ref = flatpak_dir_current_ref (dir, app_id, cancellable);
+          if (current_ref != NULL)
+            break;
+        }
+    }
+
+  if (current_ref)
+    return g_steal_pointer (&current_ref);
+
+  flatpak_fail_error (error, FLATPAK_ERROR_NOT_INSTALLED, _("%s not installed"), app_id);
```

Comment:
I think you need to add this file to POTFILES.in

Reply:
Good catch. `flatpak-repo-utils.c` also needed adding.